share/pmacc/examples/gameOfLife2D/include/Evolution.hpp

```diff
@@ -30,140 +30,265 @@
-                /* get position in grid in units of SuperCells from blockID */
-                const Space block(mapper.getSuperCellIndex(Space(blockIdx)));
-                /* convert position in unit of cells */
-                const Space blockCell = block * Mapping::SuperCellSize::toRT();
-                /* convert CUDA dim3 to DataSpace<DIM3> */
-                const Space threadIndex(threadIdx);
-                const uint32_t cellIdx = DataSpaceOperations<DIM2>::map(
-                        mapper.getGridSuperCells() * Mapping::SuperCellSize::toRT(),
-                        blockCell + threadIndex);
-
-                /* get uniform random number from seed  */
-                auto rng = nvidia::rng::create(
-                                    nvidia::rng::methods::Xor(seed, cellIdx),
-                                    nvidia::rng::distributions::Uniform_float());
-
-                /* write 1(white) if uniform random number 0<rng<1 is smaller than 'fraction' */
-                buffWrite(blockCell + threadIndex) = (rng() <= fraction);
-            }
-        };
-    }
-
-    template<class MappingDesc>
+            using namespace mappings::threads;
+
+            using SuperCellSize = typename T_Mapping::SuperCellSize;
+            constexpr uint32_t cellsPerSuperCell = pmacc::math::CT::volume< SuperCellSize >::type::value;
+            constexpr uint32_t numWorkers = T_numWorkers;
+            uint32_t const workerIdx = threadIdx.x;
+
+            // get position in grid in units of SuperCells from blockID
+            Space const block( mapper.getSuperCellIndex( Space( blockIdx ) ) );
+            // convert position in unit of cells
+            Space const blockCell = block * T_Mapping::SuperCellSize::toRT( );
+            // convert CUDA dim3 to DataSpace<DIM3>
+            Space const threadIndex(threadIdx);
+
+            uint32_t const globalUniqueId = DataSpaceOperations< DIM2 >::map(
+                mapper.getGridSuperCells() * T_Mapping::SuperCellSize::toRT(),
+                blockCell + DataSpaceOperations< DIM2 >::template map< SuperCellSize >( workerIdx )
+            );
+
+            // get uniform random number from seed
+            auto rng = nvidia::rng::create(
+                nvidia::rng::methods::Xor( seed, globalUniqueId ),
+                nvidia::rng::distributions::Uniform_float( )
+            );
+
+            ForEachIdx<
+                IdxConfig<
+                    cellsPerSuperCell,
+                    numWorkers
+                >
+            >{ workerIdx }(
+                [&](
+                    uint32_t const linearIdx,
+                    uint32_t const
+                )
+                {
+                    // cell index within the superCell
+                    DataSpace< DIM2 > const cellIdx = DataSpaceOperations< DIM2 >::template map< SuperCellSize >( linearIdx );
+                    // write 1(white) if uniform random number 0<rng<1 is smaller than 'threshold'
```

Comment:
I know, this is nitpicking, but is `0<rng<1` really true? So `rng` can't be `0` or `1`? I am just curious, never worked with `nvidia::rng`